<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>View</class>
 <widget class="QMainWindow" name="View">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>918</width>
    <height>904</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>570</width>
    <height>520</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>5700</width>
    <height>5200</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MLP</string>
  </property>
  <widget class="QWidget" name="Paint">
   <widget class="QTabWidget" name="tabWidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>900</width>
      <height>891</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>671</width>
      <height>641</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>900</width>
      <height>900</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::North</enum>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Rounded</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="elideMode">
     <enum>Qt::ElideRight</enum>
    </property>
    <widget class="QWidget" name="tab1">
     <attribute name="title">
      <string>Settings</string>
     </attribute>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>148</y>
        <width>855</width>
        <height>16</height>
       </rect>
      </property>
      <property name="lineWidth">
       <number>2</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBoxTrain">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>155</y>
        <width>911</width>
        <height>701</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="buttonTrain">
       <property name="geometry">
        <rect>
         <x>760</x>
         <y>30</y>
         <width>111</width>
         <height>91</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Train</string>
       </property>
      </widget>
      <widget class="QCustomPlot" name="widget" native="true">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>140</y>
         <width>491</width>
         <height>531</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonChooseTraining">
       <property name="geometry">
        <rect>
         <x>640</x>
         <y>30</y>
         <width>110</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
        </font>
       </property>
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
       <property name="text">
        <string>Choose</string>
       </property>
      </widget>
      <widget class="QTextBrowser" name="textTerminal">
       <property name="geometry">
        <rect>
         <x>540</x>
         <y>200</y>
         <width>331</width>
         <height>471</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>35</y>
         <width>121</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="text">
        <string>File for training</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineeditFileNameTraining">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>35</y>
         <width>481</width>
         <height>32</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonChooseTesting">
       <property name="geometry">
        <rect>
         <x>640</x>
         <y>80</y>
         <width>110</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Choose</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>85</y>
         <width>121</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="text">
        <string>File for testing</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineeditFileNameTest">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>85</y>
         <width>481</width>
         <height>32</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonSaveWeight">
       <property name="geometry">
        <rect>
         <x>640</x>
         <y>130</y>
         <width>110</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Save Weight</string>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonLoadWeight">
       <property name="geometry">
        <rect>
         <x>760</x>
         <y>130</y>
         <width>110</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Load Weight</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBoxSettings">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>0</y>
        <width>911</width>
        <height>157</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>30</y>
         <width>231</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="text">
        <string>The number of hidden layers:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>455</x>
         <y>60</y>
         <width>221</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="text">
        <string>The number of groups:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>90</y>
         <width>231</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Test sample size:</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboboxLayersNumber">
       <property name="geometry">
        <rect>
         <x>305</x>
         <y>30</y>
         <width>71</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <item>
        <property name="text">
         <string>2</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>3</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>4</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>5</string>
        </property>
       </item>
      </widget>
      <widget class="QSpinBox" name="spinboxNumberOfGroups">
       <property name="geometry">
        <rect>
         <x>700</x>
         <y>60</y>
         <width>59</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinboxNumberOfEpoch">
       <property name="geometry">
        <rect>
         <x>310</x>
         <y>60</y>
         <width>59</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="value">
        <number>5</number>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBoxCrossValidationMode">
       <property name="geometry">
        <rect>
         <x>450</x>
         <y>30</y>
         <width>271</width>
         <height>20</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="layoutDirection">
        <enum>Qt::RightToLeft</enum>
       </property>
       <property name="text">
        <string>Cross-validation mode:                 </string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doublespinboxSampleSize">
       <property name="geometry">
        <rect>
         <x>310</x>
         <y>90</y>
         <width>59</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>60</y>
         <width>231</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="text">
        <string>The number of epoch:</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radiobuttonGraphImpl">
       <property name="geometry">
        <rect>
         <x>800</x>
         <y>93</y>
         <width>81</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Graph</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="radiobuttonMatrixImpl">
       <property name="geometry">
        <rect>
         <x>700</x>
         <y>93</y>
         <width>81</width>
         <height>22</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>18</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Matrix</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>455</x>
         <y>93</y>
         <width>231</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Perceptron implementation:</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab3">
     <attribute name="title">
      <string>Demonstration</string>
     </attribute>
     <widget class="QGroupBox" name="groupBoxDemonstration">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>0</y>
        <width>911</width>
        <height>861</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="buttonLoadPic">
       <property name="geometry">
        <rect>
         <x>760</x>
         <y>70</y>
         <width>110</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
        </font>
       </property>
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
       <property name="text">
        <string>Load pic.</string>
       </property>
      </widget>
      <widget class="QGraphicsView" name="graphicsView">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>70</y>
         <width>540</width>
         <height>540</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="baseSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:black;</string>
       </property>
      </widget>
      <widget class="QLabel" name="labelGuess">
       <property name="geometry">
        <rect>
         <x>600</x>
         <y>580</y>
         <width>281</width>
         <height>181</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>116</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>680</x>
         <y>510</y>
         <width>110</width>
         <height>58</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>22</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>I suppose
it:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
      </widget>
      <widget class="QGraphicsView" name="graphicsviewLoadPic">
       <property name="geometry">
        <rect>
         <x>590</x>
         <y>140</y>
         <width>291</width>
         <height>291</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="baseSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color:black;</string>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonCleanGraphicsView">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>700</y>
         <width>111</width>
         <height>91</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Clean</string>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonGuess">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>700</y>
         <width>111</width>
         <height>91</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Guess</string>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonDelPic">
       <property name="geometry">
        <rect>
         <x>860</x>
         <y>120</y>
         <width>30</width>
         <height>34</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Arial</family>
         <pointsize>20</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
       <property name="styleSheet">
        <string notr="true">color:Red</string>
       </property>
       <property name="text">
        <string>X</string>
       </property>
      </widget>
      <widget class="Line" name="line_3">
       <property name="geometry">
        <rect>
         <x>590</x>
         <y>470</y>
         <width>291</width>
         <height>20</height>
        </rect>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">View/qcustomplot_lib/qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>comboboxLayersNumber</tabstop>
  <tabstop>spinboxNumberOfEpoch</tabstop>
  <tabstop>doublespinboxSampleSize</tabstop>
  <tabstop>checkBoxCrossValidationMode</tabstop>
  <tabstop>spinboxNumberOfGroups</tabstop>
  <tabstop>radiobuttonMatrixImpl</tabstop>
  <tabstop>radiobuttonGraphImpl</tabstop>
  <tabstop>lineeditFileNameTraining</tabstop>
  <tabstop>buttonChooseTraining</tabstop>
  <tabstop>lineeditFileNameTest</tabstop>
  <tabstop>buttonChooseTesting</tabstop>
  <tabstop>buttonTrain</tabstop>
  <tabstop>buttonSaveWeight</tabstop>
  <tabstop>buttonLoadWeight</tabstop>
  <tabstop>textTerminal</tabstop>
  <tabstop>graphicsView</tabstop>
  <tabstop>buttonLoadPic</tabstop>
  <tabstop>graphicsviewLoadPic</tabstop>
  <tabstop>buttonDelPic</tabstop>
  <tabstop>buttonCleanGraphicsView</tabstop>
  <tabstop>buttonGuess</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
